src/zenml/steps/base_step.py

```diff
@@ -260,6 +261,7 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
 
         kwargs = {**self.INSTANCE_CONFIGURATION, **kwargs}
         name = kwargs.pop(PARAM_STEP_NAME, None) or self.__class__.__name__
+        source_code = kwargs.pop(PARAM_SOURCE_CODE, None)
```

Comment:
This does not work for class-based steps now right? You can take a look at `BaseStep.caching_parameters` to see how we handle it for both

Reply:
Changed it to also work for class-based steps. This makes it much cleaner in general, great suggestion 👍